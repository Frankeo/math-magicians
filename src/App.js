import React from 'react';
import logo from './media/logo.png';
import Calculator from './components/Calculator';
import Clock from './components/Clock';

const App = () => (
  <>
    <div className="app">
      <img
        src={logo}
        alt="Logo"
        className="logo"
      />
      <div className="app-container">
        <Clock />
        <h2 className="app-header">Lets do some math!</h2>
        <Calculator />
      </div>
    </div>
  </>

);

export default App;
