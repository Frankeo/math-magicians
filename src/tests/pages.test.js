import React from 'react';
import renderer from 'react-test-renderer';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import NavbarComponent from '../components/Navbar';
import App from '../App';

describe('pages test snapshots', () => {
  it('testing Navbar component', () => {
    const tree = renderer
      .create(
        <Router basename={process.env.PUBLIC_URL}>
          <NavbarComponent />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing home page', () => {
    render(<App />);
    const Home = screen.getByText('Home');
    fireEvent.click(Home);
    expect(screen.getByText('Welcome to our page')).toMatchSnapshot();
  });
  it('testing Quote page', () => {
    render(<App />);
    const Quote = screen.getByText('Quote');
    fireEvent.click(Quote);
    expect(screen.getByText('Get motivated with new quote')).toMatchSnapshot();
  });
  it('testing Quote page', () => {
    render(<App />);
    const Calculator = screen.getByText('Calculator');
    fireEvent.click(Calculator);
    expect(screen.getByText('Lets do some math!')).toMatchSnapshot();
  });
  it('testing To do page', () => {
    render(<App />);
    const ToDo = screen.getByText('To Do');
    fireEvent.click(ToDo);
    expect(screen.getByPlaceholderText('Add Todo...')).toMatchSnapshot();
  });
});
