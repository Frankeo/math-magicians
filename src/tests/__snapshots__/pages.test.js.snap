// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages test snapshots testing Navbar component 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      <img
        alt="Logo"
        className="logo"
        src="logo.png"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="nav-links navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
        <a
          href="/todolist"
          onClick={[Function]}
        >
          To Do
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`pages test snapshots testing Quote page 1`] = `
<h2>
  Get motivated with new quote
</h2>
`;

exports[`pages test snapshots testing Quote page 2`] = `
<h2
  class="app-header"
>
  Lets do some math!
</h2>
`;

exports[`pages test snapshots testing To do page 1`] = `
<input
  class="input-text"
  name="title"
  placeholder="Add Todo..."
  type="text"
  value=""
/>
`;

exports[`pages test snapshots testing home page 1`] = `
<h1>
  Welcome to our page
</h1>
`;
